src/components/network-map-tab.js

```diff
@@ -804,31 +813,52 @@ export const NetworkMapTab = ({
         if (refIsMapManualRefreshEnabled.current && isInitialized) {
             return;
         }
+        // as long as rootNodeId is not set, we don't fetch any geodata
+        if (!rootNodeId) {
+            return;
+        }
         // Hack to avoid reload Geo Data when switching display mode to TREE then back to MAP or HYBRID
         // TODO REMOVE LATER
         if (!reloadMapNeeded) {
             return;
         }
-        if (!isInitialized) {
+        if (!isRootNodeDataLoaded) {
+            // load default node map equipments
             loadMapEquipments();
-            loadAllGeoData();
+            // load root node geodata
+            loadRootNodeGeoData();
         } else {
             updateMapEquipmentsAndGeoData();
         }
-        setInitialized(true);
         // Note: studyUuid and dispatch don't change
     }, [
+        rootNodeId,
         disabled,
         studyUuid,
         currentNode,
         loadMapEquipments,
         updateMapEquipmentsAndGeoData,
-        loadAllGeoData,
+        loadRootNodeGeoData,
+        isRootNodeDataLoaded,
         isInitialized,
         reloadMapNeeded,
         updatedSubstationsIds,
     ]);
 
+    useEffect(() => {
+        // when root node geodata are loaded, we fetch current node missing geo-data
+        // we check if equipments are done initializing because they are checked to fetch accurat missing geo data
+        if (isRootNodeDataLoaded && isEquipmentsInitiliazed && !isInitialized) {
```

Comment:
isInitialized could be renamed to isMissingGeoDataLoaded
and 
isRootNodeDataLoaded to isRootNodeGeoDataLoaded

Reply:
I agree for root node name, I will rename isInitialized to isDefaultNodeMissingGeoDataLoaded though, to be more accurate